larpandoracontent/LArHelpers/LArMCParticleHelper.cc

```diff
@@ -810,7 +865,7 @@ bool LArMCParticleHelper::PassMCParticleChecks(const MCParticle *const pOriginal
     if (NEUTRON == std::abs(pThisMCParticle->GetParticleId()))
         return false;
 
-    if ((PHOTON == pThisMCParticle->GetParticleId()) && (PHOTON != pOriginalPrimary->GetParticleId()) && (E_MINUS != std::abs(pOriginalPrimary->GetParticleId())))
+    if ((PHOTON == pThisMCParticle->GetParticleId()) && (PHOTON != GetPrimaryMCParticle(pThisMCParticle)->GetParticleId()) && (E_MINUS != std::abs(GetPrimaryMCParticle(pThisMCParticle)->GetParticleId())))
```

Comment:
I would take the opportunity to assign the result of `GetPrimaryMCParticle(pThisMCParticle)->GetParticleId()` to a variable before its use in the conditional, otherwise you're repeating exactly the same function call twice - assuming the compiler can't optimise this away for you (it probably can't).

Reply:
so, i have changed this back to pOriginalPrimary and just made sure that pOriginalPrimary is obtained in the calling function by the function GetPrimaryMCParticle instead of from the map (which is now not necessarily) the MCParticle -> primary MCParticle map

Reply:
I wondered about the details behind this change - so that's back as per the original implementation, now?

Reply:
yes, it i have put back to the original